gemelli/factorization.py

```diff
@@ -401,86 +610,103 @@ def RTPM(TE, r, ninit, nitr):
+    tol_rtpm : flat, optional
+        The minimization -- convergence break point for
+        RTPM.
 
     Returns
     -------
-    S0 : array-like
-        The eigenvalues of the factorizations
-    U : list of array-like
-        The `i`-th entry of U corresponds to
-        the factors along the `i`-th mode of TE
+    eigvals : array-like
+        The eigvals of the factorizations
+    loadings : list of array-like
+        The `i`-th entry of initial_loadings corresponds to
+        the factors along the `i`-th mode of tensor
 
     References
     ----------
-    .. [1] A. Anandkumar, R. Ge, D. Hsu,
-            S. M. Kakade, M. Telgarsky,
-            Tensor Decompositions for Learning
-            Latent Variable Models
-            (A Survey for ALT).
-            Lecture Notes in
-            Computer Science
-            (2015), pp. 19–38.
     .. [2] A. Anandkumar, R. Ge, M., Janzamin,
             Guaranteed Non-Orthogonal Tensor
             Decomposition via Alternating Rank-1
             Updates. CoRR (2014),
             pp. 1-36.
-    .. [3] P. Jain, S. Oh, in Advances in Neural
-            Information Processing Systems
-            27, Z. Ghahramani, M. Welling,
-            C. Cortes, N. D. Lawrence,
-            K. Q. Weinberger, Eds.
-            (Curran Associates, Inc., 2014),
-            pp. 1431–1439.
 
     """
-    dims = TE.shape
-    U = [np.zeros((n, r)) for n in dims]
-    S0 = np.zeros((r, 1))
-    for i in range(r):
-        tU = [np.zeros((n, ninit)) for n in dims]
-        tS = np.zeros((ninit, 1))
-        for init in range(ninit):
-            initializations = RTPM_single(
-                TE - CPcomp(S0, U), max_iter=nitr)
-
-            for tUn_idx, tUn in enumerate(tU):
-                tUn[:, init] = initializations[tUn_idx]
-                tUn[:, init] = tUn[:, init] / norm(tUn[:, init])
-
-            tS[init] = TenProjAlt(TE - CPcomp(S0, U),
-                                  [tUn[:, [init]] for tUn in tU])
-
-        idx = np.argmin(tS, axis=0)[0]
-
-        for tUn, Un in zip(tU, U):
-            Un[:, i] = tUn[:, idx] / norm(tUn[:, idx])
-
-        S0[i] = TenProjAlt(TE - CPcomp(S0, U),
-                           [Un[:, [i]] for Un in U])
-
-    return S0, U
-
-
-def RTPM_single(tensor, max_iter=50):
+    
+    # tensor shape is the number of loadings
+    dims = tensor.shape
+    # for each dim. initalize a loading fiber
+    loadings = [np.ones((n, n_components)) for n in dims]
+    # initlize n_component eigvals
+    eigvals = np.ones((n_components, 1))
+    # iterate on loadings from first to last
+    for r in range(n_components):
+        # loadings and eigs to fille for
+        # each random initalization vector
+        tU = [np.zeros((n, n_initializations)) for n in dims]
+        tS = np.zeros((n_initializations, 1))
+        for init in range(n_initializations):
+            # random initialization
+            if random_state is None or isinstance(random_state, int):
```

Comment:
Should the random seed be set before this for loop begins?  Won't this set each random intialization to the same thing?